pandas/core/dtypes/common.py

```diff
@@ -1623,6 +1623,8 @@ def pandas_dtype(dtype) -> DtypeObj:
         return dtype.dtype
     elif isinstance(dtype, (np.dtype, ExtensionDtype)):
         return dtype
+    elif "list" in str(dtype) and "pyarrow" in str(dtype):
```

Comment:
This fix is too specific and probably doesn't fix the core issue

Reply:
@mroeschke I second this. A more general fix is needed.